bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ChannelItemProvider.java

```diff
@@ -175,6 +174,14 @@ public void run() {
         }
     }
 
+    private void initialize() {
+        logger.debug("Enabling channel item provider.");
+        initialized = true;
```

Comment:
Isn't that the case we need to use volatile?

The variable initialized could be written by a single thread executor created and read by another thread.